dbms/src/Debug/dbgKVStore/dbgFuncInvestigator.cpp

```diff
@@ -0,0 +1,287 @@
+            result.in_lock.emplace_back(region_id);
+    }
+    result.regions = regions;
+    output(fmt::format("find key result {}", result.toString()));
+}
+
+BlockInputStreamPtr dbgFuncFindKeyDt(Context & context, const ASTs & args)
+{
+    if (args.size() < 4)
+        throw Exception(
+            "Args not matched, should be: database-name, table-name, key, value, [key2, value2, ...]",
+            ErrorCodes::BAD_ARGUMENTS);
+
+    auto & tmt = context.getTMTContext();
+    const String & database_name_raw = typeid_cast<const ASTIdentifier &>(*args[0]).name;
+    const String & table_name_raw = typeid_cast<const ASTIdentifier &>(*args[1]).name;
+
+    auto maybe_database_name = mappedDatabaseWithOptional(context, database_name_raw);
+    if (maybe_database_name == std::nullopt)
+    {
+        LOG_INFO(DB::Logger::get(), "Can't find database {}", database_name_raw);
+        return std::make_shared<StringStreamBlockInputStream>("Error");
+    }
+    auto mapped_database_name = maybe_database_name.value();
+    auto mapped_qualified_table_name = mappedTable(context, database_name_raw, table_name_raw);
+    auto mapped_table_name = mapped_qualified_table_name.second;
+
+    auto schema_syncer = tmt.getSchemaSyncerManager();
+    auto storage = tmt.getStorages().getByName(mapped_database_name, table_name_raw, false);
+    if (storage == nullptr)
+    {
+        LOG_INFO(DB::Logger::get(), "Can't find database and table {}.{}", mapped_database_name, mapped_table_name);
+        return std::make_shared<StringStreamBlockInputStream>("Error");
+    }
+
+    auto table_info = storage->getTableInfo();
+    schema_syncer->syncTableSchema(context, table_info.keyspace_id, table_info.id);
+    if (table_info.partition.num > 0)
+    {
+        for (const auto & def : table_info.partition.definitions)
+        {
+            schema_syncer->syncTableSchema(context, table_info.keyspace_id, def.id);
+        }
+    }
+
+    String key = safeGet<String>(typeid_cast<const ASTLiteral &>(*args[2]).value);
+    String value = safeGet<String>(typeid_cast<const ASTLiteral &>(*args[3]).value);
+
+    constexpr size_t OFFSET = 4;
+    FmtBuffer fmt_buf;
+    auto key_size = args.size() - OFFSET;
+    if (key_size & 1)
+    {
+        LOG_INFO(DB::Logger::get(), "Key-values should be in pair {}", database_name_raw, table_name_raw);
+        return std::make_shared<StringStreamBlockInputStream>("Error");
+    }
+    for (size_t i = 0; i != key_size / 2; i++)
+    {
+        String k = safeGet<String>(typeid_cast<const ASTLiteral &>(*args[OFFSET + 2 * i]).value);
+        String v = safeGet<String>(typeid_cast<const ASTLiteral &>(*args[OFFSET + 2 * i + 1]).value);
+        fmt_buf.fmtAppend(" and {} = {}", k, v);
+    }
+    auto query = fmt::format(
+        "selraw *,_INTERNAL_VERSION,_INTERNAL_DELMARK,_tidb_rowid from {}.{} where {} = {}{}",
+        mapped_database_name,
+        mapped_table_name,
+        key,
+        value,
+        fmt_buf.toString());
```

Comment:
Will it cause error when the table pk_is_handle is true and no _tidb_rowid?

Reply:
fixed